src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -622,6 +625,135 @@ private void LogExceptionEvents(List<ExecutionException> exceptions, CompilerDat
             return sb;
         }
 
+        private StringBuilder ValidateChunkedEvents(string expectedEventName,
+                                                    List<ITelemetry> telemetryGeneratedEvents,
+                                                    string expectedChunkedContent,
+                                                    int chunkNumber)
+        {
+            var sb = new StringBuilder();
+            string assemblyEventsId = string.Empty;
+            string commandLineEventsId = string.Empty;
+
+            var compilerEvents = new List<EventTelemetry>();
+            var assemblyReferencesEvents = new List<EventTelemetry>();
+            var commandLineEvents = new List<EventTelemetry>();
+            var summaryEvents = new List<EventTelemetry>();
+
+            foreach (EventTelemetry telemetryEvent in telemetryGeneratedEvents)
+            {
+                switch (telemetryEvent.Name)
+                {
+                    case CompilerDataLogger.CompilerEventName:
+                        compilerEvents.Add(telemetryEvent);
+
+                        if (commandLineEventsId.Equals(string.Empty) && telemetryEvent.Properties.ContainsKey("commandLineId"))
+                        {
+                            commandLineEventsId = telemetryEvent.Properties["commandLineId"];
+                        }
+
+                        if (assemblyEventsId.Equals(string.Empty) && telemetryEvent.Properties.ContainsKey("assemblyReferencesId"))
+                        {
+                            assemblyEventsId = telemetryEvent.Properties["assemblyReferencesId"];
+                        }
+
+                        if (expectedEventName == CompilerDataLogger.AssemblyReferencesEventName && !telemetryEvent.Properties.ContainsKey("assemblyReferencesId"))
+                        {
+                            sb.AppendLine("Compiler Event is missing the `assemblyReferencesId`");
+                        }
+                        else if (expectedEventName == CompilerDataLogger.CommandLineEventName && !telemetryEvent.Properties.ContainsKey("commandLineId"))
+                        {
+                            sb.AppendLine("Compiler Event is missing the `commandLineId`");
+                        }
+                        break;
+
+                    case CompilerDataLogger.AssemblyReferencesEventName:
+                        assemblyReferencesEvents.Add(telemetryEvent);
+
+                        if (expectedEventName == CompilerDataLogger.AssemblyReferencesEventName
+                            && !expectedChunkedContent.Contains(telemetryEvent.Properties["chunkedassemblyReferences"]))
+                        {
+                            sb.AppendLine(string.Format("Unexpected {0} chunked content: {1}",
+                                CompilerDataLogger.AssemblyReferencesEventName,
+                                telemetryEvent.Properties[$"{CompilerDataLogger.AssemblyReferencesEventName}Id"]));
```

Comment:
>CompilerDataLogger.AssemblyReferencesEventName [](http://example.com/codeflow?start=61&length=46)

Why are you using this property name construction technique here but not elsewhere? Should 'assemblyReferencesId' and other constants of this kind be declared as constants, rather than embedded literals?